sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -88,6 +88,50 @@ raw_sends(__ESIMD_NS::simd<T1, n1> msgDst, __ESIMD_NS::simd<T2, n2> msgSrc0,
       msgDesc, msgSrc0.data(), msgSrc1.data(), msgDst.data());
 }
 
+/// Raw sends.  "s" suffix designates "split" variant - i.e. two sources.
+///
+/// @tparam execSize is the execution size.
+/// @tparam numSrc0 is the number of GRFs for source-0.
+/// @tparam numSrc1 is the number of GRFs for source-1.
+/// @tparam numDst is the number of GRFs for destination.
+/// @tparam sfid is the shared function ID.
+/// @tparam isEOT is the flag that indicates whether this is an EOT message
+/// (optional - default to 0).
+/// @tparam isSendc is the flag that indicates whether sendc should be used
+/// (optional - default to 0).
+/// @param msgDst is the old value of the destination operand.
+/// @param msgSrc0 is the first source operand of send message.
+/// @param msgSrc1 is the second source operand of send message.
+/// @param exDesc is the extended message descriptor.
+/// @param msgDesc is the message descriptor.
+/// @param mask is the predicate to specify enabled channels (optional - default
+/// to on).
+/// @return the vector value read from memory.
+template <uint8_t execSize, uint8_t numSrc0, uint8_t numSrc1, uint8_t numDst,
+          uint8_t sfid, uint8_t isEOT = 0, uint8_t isSendc = 0, typename T1,
+          int n1, typename T2, int n2, typename T3, int n3, int N = 16>
```

Comment:
The change looks good. I have only one last uncertainty regarding the order/number of template parameters.
- What was the motivation for splitting 1 'modifier' parameter from non-template argument to 2 parameters: isEOT+isSendc ? Wouldn't it be better to have it as close to the old/non-template variant? Was that split done because it is requested by users?

Reply:
yeah sorry let me do that, it's left over from the previous version of the PR and I missed it

Reply:
I just pushed a commit making the template arg order match the non-template arg order, but about isEOT+isSendc, I didn't split it, the existing user API already has two arguments, ex:
```
template <typename T1, int n1, typename T2, int n2, int N = 16>
__ESIMD_API void
raw_sends(__ESIMD_NS::simd<T1, n1> msgSrc0, __ESIMD_NS::simd<T2, n2> msgSrc1,
          uint32_t exDesc, uint32_t msgDesc, uint8_t execSize, uint8_t sfid,
          uint8_t numSrc0, uint8_t numSrc1, uint8_t isEOT = 0,
          uint8_t isSendc = 0, __ESIMD_NS::simd_mask<N> mask = 1) {
```

```
template <uint8_t execSize, uint8_t sfid, uint8_t numSrc0, uint8_t numSrc1,
          uint8_t isEOT = 0, uint8_t isSendc = 0, typename T1, int n1,
          typename T2, int n2, int N = 16>
__ESIMD_API void raw_sends(__ESIMD_NS::simd<T1, n1> msgSrc0,
                           __ESIMD_NS::simd<T2, n2> msgSrc1, uint32_t exDesc,
                           uint32_t msgDesc,
                           __ESIMD_NS::simd_mask<N> mask = 1) {
```

You can see both have seperate isEOT and isSendc arguments.

Reply:
Indeed. I had some glitch, or looked at wrong place (maybe at __esimd_...raw_send.. call). Yes, the old API already had 2 params instead of 1-modifier.
Thank you for patience.

Reply:
yeah the intrinsic has a single arg, but the user-visible function has two, a little confusing :)